### PR TITLE
WT-4991 Migrate Jenkins wiredtiger-test-unit-zseries job to Evergreen

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
Added a new buildvariant "ubuntu1804-zseries-test" to support zseries unit test.

### CODE REVIEW

**Comment by raviprakashgiri29** [comment]
Evergreen patch build path below

https://evergreen.mongodb.com/version/5dcc8fc99ccd4e114cc38418

File: test/evergreen.yml [hunk 1]
```diff
@@ -1766,3 +1766,17 @@ buildvariants:
   tasks:
   - name: compile
   - name: unit-test
+
+- name: ubuntu1804-zseries
+  display_name: Ubuntu 18.04 ZSeries
+  run_on:
+  - ubuntu1804-zseries-test
+  expansions:
+    configure_env_vars: PATH=/opt/mongodbtoolchain/v3/bin:$PATH
```
Comment (quchenhao) [inline test/evergreen.yml]: Why we are not building with mongodb toolchain here

Comment (raviprakashgiri29) [reply test/evergreen.yml]: I have updated the "configure_env_vars" to set PATH and CFLAGS.

File: test/evergreen.yml [hunk 2]
```diff
@@ -1758,11 +1758,25 @@ buildvariants:
   run_on:
   - ubuntu1804-power8-test
   expansions:
-    configure_env_vars: PATH=/opt/mongodbtoolchain/v3/bin:$PATH
+    configure_env_vars: PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC"
```
Comment (quchenhao) [inline test/evergreen.yml]: We should add CC and CXX flags as well.

Comment (raviprakashgiri29) [reply test/evergreen.yml]: Updated the configure_env_vars to use mongodb toolchain binaries

Comment (lukech) [reply test/evergreen.yml]: Agreed. @raviprakashgiri29 would you mind adding those missing env settings (PATH, CC, CXX) for `macos-1012` build variant, `configure-combinations` task to make the `configure_env_vars` setting consistent within evergreen.yml?

File: test/evergreen.yml [hunk 3]
```diff
@@ -1758,11 +1758,25 @@ buildvariants:
...
-    posix_configure_flags: --enable-silent-rules --enable-diagnostic --enable-python --enable-zlib --enable-snappy --enable-strict --enable-static --prefix=$(pwd)/LOCAL_INSTALL CFLAGS="-ggdb -fPIC"
+    posix_configure_flags: --enable-silent-rules --enable-diagnostic --enable-python --enable-zlib --enable-snappy --enable-strict --enable-static --prefix=$(pwd)/LOCAL_INSTALL
     test_env_vars: PATH=/opt/mongodbtoolchain/v3/bin:$PATH LD_LIBRARY_PATH=$(pwd)/.libs top_srcdir=$(pwd)/.. top_builddir=$(pwd)
     make_command: PATH=/opt/mongodbtoolchain/v3/bin:$PATH make
   tasks:
   - name: compile
   - name: unit-test
+
+- name: ubuntu1804-zseries
+  display_name: Ubuntu 18.04 ZSeries
```
Comment (lukech) [inline test/evergreen.yml]: `ZSeries` --> `zSeries` please.

File: test/evergreen.yml (hunk 2 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: Yes, I agree. I have updated configure_env_vars setting consistent across evergreen.yml.

File: test/evergreen.yml (hunk 3 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: Fixed :)

**Comment by raviprakashgiri29** [comment]
Updated OS X 10.12 configure_env_vars not to use mongodbtool chain binaries as it was failing with mongodbtoolchain

**Comment by raviprakashgiri29** [comment]
> Thanks for making the changes to line up the environment variable setting for the configure command across build variants. It's okay to skip macos for this change, would you mind writing a ticket so that we won't lose track of the issue - my understanding is mongodbtoolchain should be able to work under macos environment.

Created a jira ticket WT-5248 to track macos mongodbtoolchain issue.

**Comment by raviprakashgiri29** [comment]
Created a jira ticket [WT-5248]( https://jira.mongodb.org/browse/WT-5248 ) to track macos mongodbtoolchain issue.

**Pull request merged**